pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,513 @@
+	multiArchValidator.validate()
+
+	for _, err := range multiArchValidator.Warns {
+		result.Add(errors.WarnFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+// validateContainerTool verifies if the container tool informed is valid
+func validateContainerTool(containerTool string) (string, error) {
+	if len(containerTool) == 0 || containerTool == "none" {
+		containerTool = "docker"
+	} else if containerTool != "docker" && containerTool != "podman" {
+		return containerTool, fmt.Errorf("invalid value (%s) for (%s). One of: [docker, podman, none] "+
+			"(If not set, the default value is docker)", ContainerToolsKey, containerTool)
+	}
+	return containerTool, nil
+}
+
+// validate performs all required checks to validate the bundle against the Multiple Architecture
+// configuration to guess the missing labels and/or highlight what are the missing Architectures
+// for the images (for what is configured to be supported AND for what we guess that is supported
+// and just is missing a label).
+func (data *MultiArchValidator) validate() {
+	data.AllArchFound = make(map[string]string)
+	data.AllOsFound = make(map[string]string)
+
+	data.getInfraLabelsFromCSV()
+	data.getImagesFromCSV()
+	data.inspectAllImages()
+	data.getAllPossibleArchSupported()
+	data.checks()
+}
+
+// getInfraLabelsFromCSV will gather the respective labels from the CSV
+func (data *MultiArchValidator) getInfraLabelsFromCSV() {
+	for k, v := range data.Bundle.CSV.ObjectMeta.Labels {
+		if strings.Contains(k, operatorFrameworkArchLabel) && v == "supported" {
+			data.InfraArchLabels = append(data.InfraArchLabels, k)
+		}
+	}
+	for k, v := range data.Bundle.CSV.ObjectMeta.Labels {
+		if strings.Contains(k, operatorFrameworkSoLabel) && v == "supported" {
+			data.InfraSOLabels = append(data.InfraSOLabels, k)
+		}
+	}
+}
+
+// getImagesFromCSV will add all images found in the CSV
+// it will be looking for all containers images and what is defined
+// via the spec.RELATE_IMAGE (required for disconnect support)
+func (data *MultiArchValidator) getImagesFromCSV() {
+	data.Images = make(map[string][]Platform)
+	if data.Bundle.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs != nil {
+		for _, v := range data.Bundle.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+			for _, c := range v.Spec.Template.Spec.Containers {
+				data.Images[c.Image] = append(data.Images[c.Image], Platform{})
+			}
+		}
+	}
+
+	for _, v := range data.Bundle.CSV.Spec.RelatedImages {
+		data.Images[v.Image] = append(data.Images[v.Image], Platform{})
+	}
+}
+
+// runManifestInspect executes the command for we are able to check what
+// are the Architecture(s) and SO(s) supported per each image found
+func runManifestInspect(image, tool string) (ManifestInspect, error) {
+	cmd := exec.Command(tool, "manifest", "inspect", image)
+	output, err := runCommand(cmd)
+	if err != nil || len(output) < 1 {
```

Comment:
In the case that `err` is `nil` and the length of the output is 0, you’ll get a weird silent failure. I would prefer:

```go
if err != nil {
    // return err
} else if len(output) == 0 {
    // return empty output error
}
```

The current code will return an empty manifest and nil error when the output is empty.

Reply:
For both, the return is always the same `return manifestInspect{}, err`.
Not sure if I could follow up on your suggestion.

Reply:
The situation I'm thinking of is when we have:
```go
output := ""
err := nil
```
The return value of `runManifestInspect` will be `ManifestInspect{}, nil`, which will be confusing since the caller will need to know the meaning of the `ManifestInspect` zero value. I think we would want `ManifestInspect{}, errors.New("no output returned from manifest tool")` to make it clear what happened.

Reply:
The `no output returned from manifest tool` would return an error always in this case. 
How can we do a docker manifest inspect and the manifest be parsed and empty and no error returned?

Reply:
Then why check for the empty output in the first place? From what you just said if `len(output) == 0` is `true` then `err != nil` will also be `true`.

I also don't want to hold up this PR on this relatively small edge case, so I am good to approve if you are confident that we need to check the output length and it will never be empty without an error, but I stand by that you should handle `len(output) == 0` in a separate `if/else` branch or drop the check entirely.

Reply:
I think you are right the len(output) == 0 is redundant.
Removed :-)

Reply:
I think you are right the len(output) == 0 is redundant.
Removed :-)

Reply:
I think you are right the len(output) == 0 is redundant.
Removed :-)